pywbem/mof_compiler.py

```diff
@@ -2178,6 +2191,41 @@ def CreateClass(self, *args, **kwargs):
             self.classes[self.default_namespace] = \
                 NocaseDict({cc.classname: cc})
 
+        objects = list(cc.properties.values())
+        for meth in cc.methods.values():
+            objects += list(meth.parameters.values())
+
+        for obj in objects:
+            # Validate that reference_class exists in repo
+            if obj.type == 'reference':
+                try:
+                    self.GetClass(obj.reference_class, LocalOnly=True,
+                                  IncludeQualifiers=True)
+                except CIMError as ce:
+                    if ce.args[0] == CIM_ERR_NOT_FOUND:
+                        ce.args = (CIM_ERR_INVALID_PARAMETER,
+                                   'Reference object "%s" of class "%s" '
+                                   'reference_class "%s" does not exist' %
+                                   (obj.name, cc.classname,
+                                    obj.reference_class))
+                    raise
+
+            elif obj.type == 'string':
+                if 'EmbeddedInstance' in obj.qualifiers:
+                    eiqualifier = obj.qualifiers['EmbeddedInstance']
+                    try:
+                        self.GetClass(eiqualifier.value, LocalOnly=True,
+                                      IncludeQualifiers=False)
+                    except CIMError as ce:
+                        if ce.args[0] == CIM_ERR_NOT_FOUND:
+                            ce.args = (CIM_ERR_INVALID_PARAMETER,
+                                       'EmbeddedInstance qualifier classname '
+                                       '"%s"  in object "%s" of class "%s" '
+                                       'does not exist.' %
```

Comment:
This exception text could be improved, because at a first read, it is not clear whether the qualifier has no class name or the class does not exist. Thinking about it then makes it clear, but it could be better. Also, it might be helpful to mention the namespace.
How about: "Class %r specified by the EmbeddedInstance qualifier on element %r in class %r in namespace %r does not exist."

Reply:
OK

Reply:
DONE. Revamped message and change %s to %r for template string